atd-cr3-api/server.py

```diff
@@ -425,42 +425,54 @@ def user_delete_user(id):
         abort(403)
 
 
-@APP.route("/crashes/associate_location/", methods=["PUT", "POST"])
+@APP.route("/events/", methods=["PUT", "POST"])
 def associate_location():
     # Require matching token
-    incoming_token = request.headers.get('Hasura-Event-Api')
+    incoming_token = request.headers.get("HASURA_EVENT_API")
+    incoming_event_name = request.headers.get("HASURA_EVENT_NAME", "")
     hashed_events_api = hashlib.md5()
     hashed_events_api.update(str(HASURA_EVENT_API).encode("utf-8"))
     hashed_incoming_token = hashlib.md5()
     hashed_incoming_token.update(str(incoming_token).encode("utf-8"))
 
     # Return error if token doesn't match
     if hashed_events_api.hexdigest() != hashed_incoming_token.hexdigest():
-        return {
-            "statusCode": 403,
-            "body": json.dumps({'message': 'Forbidden Request'})
-        }
+        return {"statusCode": 403, "body": json.dumps({"message": "Forbidden Request"})}
+
+    # Check if there is an event name provided, if not provide feedback.
+    if incoming_event_name == "":
+        return {"statusCode": 403, "body": json.dumps({"message": "Forbidden Request"})}
 
     # We continue the execution
     try:
-        sqs = boto3.client('sqs')
-        queue_url = HASURA_EVENTS_SQS_URL
+        sqs = boto3.client("sqs")
+        queue_url = (
+            HASURA_EVENTS_SQS_URL[0:48]
```

Comment:
can you add some commented notes on why we're trimming 48 characters? Maybe we set that as a constant with a explanatory variable name in case we need to update it in the future.

Reply:
The url is a constant that is composed by the region name, aws domain, and aws account number. We can assume the url will always follow that pattern whenever we create new queues.

Reply:
Done, made comments.